api.js

```diff
@@ -296,14 +195,141 @@ const api = {
+        bot.sendMessage({
+          to: channelId,
+          message: `<@${userId}> Nice try dipshit`
+        });
+
+        return;
+      }
+
+      const user = this.database.ref(`users/${userId}`);
+      const userLoanTo = this.database.ref(`users/${loanTo}`);
+      user.once('value', snapshot => {
+        if (snapshot.val()) {
+          if (snapshot.val().glimmer < amount) {
+            bot.sendMessage({
+              to: channelId,
+              message: `<@${userId}> you don't have that much glimmer to loan dickface.`
+            });
+            return;
+          }
+          bot.sendMessage({
+            to: channelId,
+            message: `<@${userId}> Good financial decision`
+          });
+          let glimmer = Number(snapshot.val().glimmer) - Number(amount);
+          snapshot.ref.update({ glimmer });
+        }
+      });
+
+      userLoanTo.once('value', snapshot => {
+        if (snapshot.val()) {
+          bot.sendMessage({
+            to: channelId,
+            message: `<@${loanTo}> You just got ${amount} glimmer! Good luck...`
+          });
+
+          if (!(isNaN(snapshot.val().oweTo.userId))) {
+            let owed = Number(snapshot.val().oweTo.userId) + Number(amount);
+            let oweTo = {userId: Number(owed)};
+            console.log(owed);
+            snapshot.ref.update({ oweTo });
+          }
+          else {
+            let oweTo = {};
+            oweTo[userId] = Number(amount);
+            console.log(Number(amount));
+            snapshot.ref.update({ oweTo });
+          }
+
+          let glimmer = Number(snapshot.val().glimmer) + Number(amount);
+          snapshot.ref.update({ glimmer });
+        }
+      });
+    }
+    catch (e) {
+      logger.error(`Error in loanGlimmer for ${userId} and amount ${amount}: ${e}`);
+    }
+  },
+
+  // @summary collects glimmer from a user
+  // @param userId - calling user
+  // @param collectFrom - the user to collect from
+  // @param bot - this bot, duh
+  // @channelId - id of the channel to write to
+  collect(userId, collectFrom, bot, channelId) {
+    try {
+      let paid = 0;
+
+      const user = this.database.ref(`users/${userId}`);
+      const userCollectFrom = this.database.ref(`users/${collectFrom}`);
+      userCollectFrom.once('value', snapshot => {
+        if (snapshot.val()) {
+          if (isNaN(snapshot.val().oweTo.userId)) {
+            bot.sendMessage({
+              to: channelId,
+              message: `<@${userId}> Nope`
+            });
+          }
+          else {
+            let owed = Number(snapshot.val().oweTo.userId);
+            let glimmer = (snapshot.val().glimmer > owed) ? (snapshot.val().glimmer - Math.floor(owed)) : 0;
+            paid = (snapshot.val().glimmer - glimmer);
+
+            console.log(owed);
+            console.log(paid);
+
+            snapshot.ref.update({ glimmer });
+
+            let oweTo = {};
+            oweTo[userId] =  Number((owed-paid) > 0 ? (owed - paid) : 0);
```

Comment:
If they pay it all off, we should probably delete that userId from their oweTo object right? Otherwise in the above functions when we're checking if they owe, it could still have that userId in their oweTo object but have a value of 0. 

`delete oweTo[userId]` will work